documents/description.md

```diff
@@ -2,8 +2,105 @@
 
 ## Motivation
 
+###### Why is this project useful or interesting, what problem are you trying to address, and why is a DSL an appropriate solution?
+This project is useful since it will attempt to be a tool for teaching novices programming while also challenging experienced 
+programmers in a fun game enviornment. I originally became interested in programming in 4th grade by learning how to script using
+_SCM Draft 2_ and _StarForge_, two 3rd party map editors made for creating custom scenarios for the popular computer game _Starcraft_.
+My goal is to create a similar more defined experience to help inspire new or novice programmers to learn programming while also 
+challenging experienced programmers to tackle some interesting AI and navigation problems.
+
+A DSL is an appropriate solution for this project since I want to embed the language inside of a game. Although I definetly want this
+language to feel C-like, I want to include very detailed feedback/errorchecking to the user, especially useful for new programmers,
+and an API-like interface to interact with a given spaceship. When you make a syntax error, I want a character in the game to complain
+to you about it!
+
+
+
 ## Language domain
 
+###### What is the domain that this language addresses, and why is the domain useful?
+The domain of this language is that of a game. That being said, a game is not a classification in itself. I want this game to have
+levels that are educational to programming novices while simultaniously being able to have levels that challenge even veteran
+programmers. This means the game's DSL language needs to present its syntax in an interesting way that helps build up domain knowledge
+to novices while also supporting a plethora of features neccessary for more interesting challenges. This domain is useful since it can
+be both entertaining and educational!
+
+###### Who will benefit from this language?
+As mentioned above, although they will get different things out of it, this language should ideally benefit both novice and
+experienced programmers. That is, a novice programmer should get an interactive educational and fun experience while a experienced 
+programmer should get an interactive challenge with regards to AI and naviagation.
+
```

Comment:
You don't need to create sub-sections for each prompt in the assignment. The writeup would be clearer as a cohesive whole (and you wouldn't have to repeat yourself).